### DISCUSSION TITLE
compile warning for union initializer

### DISCUSSION THREAD

**Discussion by yzgr2** [protobuf-c/protobuf-c#593] [Q&A] [answered]
There is a compile warning for union initializer if the first field the union is not a struct.
example: 

1. hey.proto 

syntax = "proto3";
package hey;
message ActionMessage {
    reserved 1;
    oneof payload {
        bytes data = 2;
        // test_msg msg = 3;
    } 
    uint32 seq = 5;
    uint32 ack = 6;
    uint32 win = 7;
}

2. generated hey.pb-c.h

struct  _Hey__ActionMessage
{
  ProtobufCMessage base;  
  uint32_t seq;
  uint32_t ack;
  uint32_t win;
  Hey__ActionMessage__PayloadCase payload_case;
  union {   
        **ProtobufCBinaryData data;**
  };
};
#define HEY__ACTION_MESSAGE__INIT \
 { PROTOBUF_C_MESSAGE_INIT (&hey__action_message__descriptor) \
    , 0, 0, 0, HEY__ACTION_MESSAGE__PAYLOAD__NOT_SET, **{0}** }

3. example.c:

 Hey__ActionMessage act = HEY__ACTION_MESSAGE__INIT;

when compiling with -Wall flags, gcc shows: 
**hey.pb-c.h:50:2: warning: missing braces around initializer [-Wmissing-braces]**

in this case, since we try to initialize the anonymous union with _{0}_, but the first field of the union is _ProtobufCBinaryData data_ , it is a structure.  the warning can be removed by changing _{0}_ to _{{0}}_

**Comment by lipnitsk** [comment, marked as answer]
Duplicate of #374
